packages/gestalt-datepicker/src/InternalDateField.js

```diff
@@ -0,0 +1,285 @@
+        <input
+          autoComplete={ownerState?.passthroughProps?.autoComplete ?? 'off'}
+          id={ownerState?.passthroughProps?.id}
+          className={styledClasses}
+          disabled={disabled}
+          ref={inputRef}
+          placeholder={placeholder}
+          value={value}
+          enterKeyHint={ownerState?.passthroughProps?.enterKeyHint}
+          inputMode="numeric"
+          readOnly={readOnly}
+          onClick={onClick}
+          onFocus={(event) => ownerState?.passthroughProps?.onFocus?.({ event, value })}
+          onBlur={(event) => ownerState?.passthroughProps?.onBlur?.({ event, value })}
+          onPaste={onPaste}
+          onChange={onChange}
+          onKeyDown={onKeyDown}
+          onMouseUp={onMouseUp}
+        />
+      </Box>
+    );
+  },
+);
+
+type CustomDateFieldProps = {
+  inputRef: {| ref: {| current: ?HTMLElement |} |},
+  slots: string,
+  slotProps: string,
+  ...
+};
+
+function CustomDateField({
+  inputRef: externalInputRef,
+  slots,
+  slotProps,
+  ...textFieldProps
+}: CustomDateFieldProps): Node {
+  return (
+    <CustomTextField
+      {...useDateField({
+        props: textFieldProps,
+        inputRef: externalInputRef,
+      })}
+    />
+  );
+}
+
+type InternalDateFieldProps = {|
+  autoComplete?: 'bday' | 'off',
+  disabled?: boolean,
+  disableRange?: 'disableFuture' | 'disablePast',
+  errorMessage?: Node,
+  helperText?: string,
+  id: string,
+  label?: string,
+  labelDisplay?: 'visible' | 'hidden',
+  maxDate?: Date,
+  minDate?: Date,
+  mobileEnterKeyHint?: 'enter' | 'done' | 'go' | 'next' | 'previous' | 'search' | 'send',
+  name?: string,
+  onBlur?: ({|
+    event: SyntheticFocusEvent<HTMLInputElement>,
+    value: string,
+  |}) => void,
+  onChange: (value: Date) => void,
+  onClearInput: () => void,
+  onError?: ({|
+    errorMessage: string,
+    value: ?Date,
+  |}) => void,
+  onFocus?: ({|
+    event: SyntheticFocusEvent<HTMLInputElement>,
+    value: string,
+  |}) => void,
+  readOnly?: boolean,
+  ref?: Element<'input'>, // eslint-disable-line react/no-unused-prop-types
+  size?: 'md' | 'lg',
+  value: ?Date,
+|};
+
+function InternalDateField({
+  autoComplete,
+  disabled = false,
+  disableRange,
+  errorMessage,
+  helperText,
+  id,
+  label,
+  labelDisplay = 'visible',
+  maxDate,
+  minDate,
+  mobileEnterKeyHint,
+  name,
+  onBlur,
+  onChange,
+  onClearInput,
+  onError,
+  onFocus,
+  readOnly = false,
+  size = 'md',
+  value,
+}: InternalDateFieldProps): Node {
+  const [focused, setFocused] = useState(false);
+
+  return (
+    <Box>
+      <label
+        className={classnames(styles.label, {
+          [styles.visuallyHidden]: labelDisplay === 'hidden',
+        })}
+        htmlFor={id}
+      >
+        <div className={styles.formLabel}>
+          <Text size="100">{label}</Text>
+        </div>
+      </label>
+      <Box position="relative" display="flex" alignItems="center">
+        <MUIDatePicker
```

Comment:
Why is this datepicker?

Reply:
Implementation comes from https://github.com/mui/mui-x/blob/v6.1.0/docs/data/date-pickers/custom-field/PickerWithBrowserField.js